src/ledger/assets/tokens.spec.ts

```diff
@@ -31,4 +31,31 @@ describe("Tokens", () => {
 			t1.sub(t2);
 		}, "subbing incompatible tokens should throw").throws;
 	});
+
+	it("compares tokens", () => {
+		expect(new Tokens([124n]).cmp(new Tokens([]))).to.eql("gt");
+		expect(new Tokens([124n]).cmp(new Tokens([124n]))).to.eql("eq");
+		expect(new Tokens([125n]).cmp(new Tokens([124n]))).to.eql(
+			"uncomparable",
+		);
+		expect(new Tokens([125n]).cmp(new Tokens([124n, 125n]))).to.eql("lt");
+		expect(new Tokens([126n]).cmp(new Tokens([124n, 125n]))).to.eql(
+			"uncomparable",
+		);
+		expect(new Tokens([126n]).cmp(new Tokens([124n, 125n]))).to.eql(
+			"uncomparable",
+		);
+		expect(
+			new Tokens([123n, 125n]).cmp(new Tokens([123n, 124n, 125n, 126n])),
+		).to.eql("lt");
+		expect(
+			new Tokens([123n, 124n]).cmp(new Tokens([122n, 123n, 124n, 125n])),
+		).to.eql("lt");
+		expect(
+			new Tokens([123n, 124n, 125n]).cmp(new Tokens([123n, 124n])),
+		).to.eql("gt");
+		expect(
+			new Tokens([122n, 123n, 124n, 125n]).cmp(new Tokens([123n, 124n])),
+		).to.eql("gt");
```

Comment:
Is sorting tested anywhere? otherwise, please add a `Tokens` constructor test that checks that the tokens are created in the right sorted order for some shuffled and also reverse sorted input array. also test that the ctor throws if an element is present twice.